### PR TITLE
Yisui/fix windows compilation

### PR DESCRIPTION
**Pull request by Toneyisnow**
Fix the header files on Windows:

1.   inet.h, sys/socket.h, netinet/in.h, -> Winsock2.h, windows.h
2.  <unistd.h> -> win/xdaglib/unistd.h
3.  <termios.h>    -> Disabled on Windows
4.  Adding getopt.h/getopt.c to provide library replacement for Windows.

### CODE REVIEW

**Comment by Sofarlemineur** [comment]
Thanks for your work !!!!

**Comment by sgaragagghu** [comment]
But windows compilation is already working

File: client/system.h [hunk 1]
```diff
@@ -50,7 +51,9 @@ static inline xdag_diff_t xdag_diff_div(xdag_diff_t p, xdag_diff_t q)
 #define xOPENSSL_ia32cap_P      OPENSSL_ia32cap_P
 #define xsha256_multi_block     sha256_multi_block
 
+#if !defined(_WIN32) && !defined(_WIN64)
 typedef long ssize_t;
```
Comment (sgaragagghu) [inline client/system.h]: why don't you remove it?

File: dnet/system.h [hunk 2]
```diff
@@ -25,6 +26,8 @@ static pthread_t pthread_invalid;
 #define SHUT_RDWR SD_BOTH
 #endif
 #else
+typedef long ssize_t;
```
Comment (sgaragagghu) [inline dnet/system.h]: is it needed here?

**Comment by sgaragagghu** [comment]
where are used getopt.h/getopt.c ? did you notice that they are LGPL licensed ?

So are you pushing a new windows wallet ?

File: win/system.c [hunk 3]
```diff
@@ -33,17 +41,23 @@ int tcgetattr(int fd, struct termios *tio)
 {
 	if(fd) return -1;
 	HANDLE hStdin = GetStdHandle(STD_INPUT_HANDLE);
+
+#if !defined(_WIN32) && !defined(_WIN64)
```
Comment (sgaragagghu) [inline win/system.c]: isn't this file supposed to be only for windows?

**Comment by sgaragagghu** [comment]
maybe you should divide the improvement to the core from the new wallet and push the new wallet in a separated repo

**Comment by Toneyisnow** [comment]
Closing this, see the new pull request from xDagWallet/yisui/winsupport.

**Pull request closed without merging**